samples/net/common/sample_app_setup.c

```diff
@@ -31,7 +31,7 @@
 #include <ieee802154_settings.h>
 #endif
 
-static struct k_sem waiter = K_SEM_INITIALIZER(waiter, 0, 1);
+static K_SEM_DEFINE(waiter, 0, 1);
```

Comment:
This file was removed few days ago, and its functionality was moved into subsys/net/lib/app/init.c